libraries/WiFi/src/WiFiGeneric.cpp

```diff
@@ -1113,6 +1126,9 @@ esp_err_t WiFiGenericClass::_eventCallback(arduino_event_t *event)
 
     } else if(event->event_id == ARDUINO_EVENT_WIFI_AP_START) {
         setStatusBits(AP_STARTED_BIT);
+        if (getStatusBits() & AP_WANT_IP6_BIT){
+            esp_netif_create_ip6_linklocal(get_esp_interface_netif(ESP_IF_WIFI_AP));
```

Comment:
I'm seeing a race condition here. `esp_netif_create_ip6_linklocal` returns `-1` because:

`get_esp_interface_netif(ESP_IF_WIFI_AP)->lwip_netif->flags` has the flag `NETIF_FLAG_UP` not set when it is called. However if I add
`delay(1)`
the flag `NETIF_FLAG_UP` is set and the link-local address is successfully assigned.

I have no clue why and I have never seen this behavior with our previous code.

Any ideas?

Reply:
I also noticed some weir stuff on the AP interface. I will report those and see what is causing it. On our end, it's as it should. We need to do this on START for the AP and CONNECT on the rest

Reply:
Oops, sorry. The race condition is at `ARDUINO_EVENT_WIFI_STA_CONNECTED`, the two codes look alike, I mixed up both. Let me copy it at the right place.